sdk/examples/wallet/13_burn_native_token.rs

```diff
@@ -42,7 +43,9 @@ async fn main() -> Result<()> {
 
         // Burn a native token
         let burn_amount = U256::from(1);
-        let transaction = account.burn_native_token(*token_id, burn_amount, None).await?;
+        let transaction = account
+            .burn(Burn::new().add_native_token(*token_id, burn_amount), None)
```

Comment:
This kinda is a little bit worse UX wise.
I'd suggest we do one of these:
- keep the original functions but they internally call `burn` with a newly constructed `Burn`
- make `burn` take a `impl Into<Burn>` and impl that for `AliasId`, `NftId`, ...

Reply:
Yeah, you're right
This is what has been done:
 - rust implementation: has a `burn(impl Into<Burn>)` which allows users, to use functionality as before. The excpepition is native token, which has to be used as  structure `NativeToken` isntead of separate parameters token_id and amount.
 - the bindings: they have old API methods that use internally the `burn()` function. Additionally to that the bindings have generic `burn()` available as well too - in the case when user would like to burn multiple outputs of different types. In case this case, the user will have to use the `Burn` structure.